Sources/SwiftKafka/KafkaPollingSystem.swift

```diff
@@ -0,0 +1,299 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the swift-kafka-gsoc open source project
+//
+// Copyright (c) 2022 Apple Inc. and the swift-kafka-gsoc project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of swift-kafka-gsoc project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import Crdkafka
+import Logging
+import NIOConcurrencyHelpers
+import NIOCore
+
+/// Our `AsyncSequence` implementation wrapping an `NIOAsyncSequenceProducer`.
+public struct KafkaAsyncSequence<Element, BackPressure: NIOAsyncSequenceProducerBackPressureStrategy>: AsyncSequence {
+    typealias PollingSystem = KafkaPollingSystem<Element, BackPressure>
+    typealias WrappedSequence = NIOAsyncSequenceProducer<Element, BackPressure, PollingSystem>
+    let wrappedSequence: WrappedSequence
+
+    /// Our `AsynceIteratorProtocol` implementation wrapping `NIOAsyncSequenceProducer.AsyncIterator`.
+    public struct KafkaAsyncIterator: AsyncIteratorProtocol {
+        let wrappedIterator: NIOAsyncSequenceProducer<Element, BackPressure, PollingSystem>.AsyncIterator
+
+        public mutating func next() async -> Element? {
```

Comment:
This probably should throw as well. From my understanding of the librd docs there are cases where the polling will throw an error

Reply:
From [the documentation](https://docs.confluent.io/platform/current/clients/librdkafka/html/rdkafka_8h.html#ad50c431e3a29d14da534db49bd0682a4), `rd_kafka_poll` should not be an issue. However, [`rd_kafka_consumer_poll`](https://docs.confluent.io/platform/current/clients/librdkafka/html/rdkafka_8h.html#a65cc6cb9bd72c4084f074af0361ceddf) can result in errors. But I think it's better if we do that in the PR dedicated for `KafkaConsumer`, right?